traefik/values.yaml

```diff
@@ -248,6 +235,23 @@ providers:
       # By default this Traefik service
       # pathOverride: ""
 
+  kubernetesGateway:
+    # -- Enable traefik GatewayClass CRD
+    enabled: false
+    ## Routes are restricted to namespace of the gateway by default.
+    ## https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.FromNamespaces
+    # namespacePolicy: All
+    # certificate:
+    #   group: "core"
+    #   kind: "Secret"
+    #   name: "mysecret"
+    # -- By default, Gateway would be created to the Namespace you are deploying Traefik to.
+    # You may create that Gateway in another namespace, setting its name below:
+    # namespace: default
+    # Additional gateway annotations (e.g. for cert-manager.io/issuer)
+    # annotations:
+    #   cert-manager.io/issuer: letsencrypt
```

Comment:
All those parameters are about configuring `Gateway` CR, not traefik gateway provider setting (detailed [here](https://doc.traefik.io/traefik/v3.1/providers/kubernetes-gateway/) in upstream).
Wdyt about :
1. Setting `Gateway` CR in a tree starting with `gateway:`
2. Setting gateway provider in a tree coming from `providers.gateway:`